[EDU-2917] refactor: review and refine Rules Engine for Edge Firewall reference (#1097)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/secure/edge-firewall/rules-engine.mdx

The Rules Engine is programmable, allowing you to define the conditions (**Criteria**) and actions (**Behaviors**). If the Criteria are met, the Behaviors will be executed.

You can use the Rules Engine for Edge Firewall to configure all operational aspects of your application's firewalls. Here are some implementations examples you can use it for:

- Blocking a request.
- Ignoring a request.
- Limiting the access rate.
- Applying a **Web Application Firewall (WAF)** policy.
- Running an Edge Function for Edge Firewall with your own security code.
- Monitoring traffic to identify threats.

An edge firewall can have as many rules as needed. These rules are interchangeable, allowing you to share the same rule across all other edge firewalls.

<LinkButton link="/en/documentation/products/secure/edge-firewall/" label="Go to Edge Firewall Reference" outlined />

<LinkButton link="/en/documentation/products/guides/secure/work-with-rules-engine/" label="Go to Work With Rules Engine Guide" outlined />

## Implementation

| Scope | Resource |
| --- | --- |

> | Secure an application | [How to secure an application](https://www.azion.com/en/documentation/products/secure/secure-application/) |

Author: ```suggestion
| Secure an application | [How to secure an application](/en/documentation/products/secure/secure-application/) |
```

---

## Description

Aside from the name of the rule, you can add a description to your rule using the **Description** field. Your description will be visible from the rule list and can be used to help you identify what the rule does.

---

## Criteria

The **Criteria** section of **Rules Engine for Edge Firewall** is where you define the conditions for executing the rule. Criteria are composed of:

- Variables
- Comparison operators
- Logic operators
- Arguments, when applicable

The inclusion of arguments in a criteria depends on the comparison operators chosen. The format of the arguments is described in the [variables](#variables) and [comparison operators](#comparison-operators) sections below. You can also add [logical operators](#logical-operators) to increase the amount of comparisons that the rule will execute.

For example, this criteria identifies requests where the URI starts with a forward slash ("/") so that the defined behavior can then be applied to them.

| | Variable | Comparison operator | Argument |
| --- | ---- | --- | --- |
| If | `Request URI` | *starts with* | `/` |

:::note 

> The availability of certain criteria depends on the enabled modules of your Edge Firewall. Go to the [Edge Firewall modules reference](/en/documentation/products/secure/edge-firewall/#how-azion-edge-firewall-works) for more details.

Author: ```suggestion
The availability of certain criteria depends on the enabled modules of your Edge Firewall. Go to the [Edge Firewall modules reference](/en/documentation/products/secure/edge-firewall/#about-edge-firewall-modules) for more details.
```

As the Rate Limit is applied by IP address of Global and by rule, if a rule has *more than one URI* specified and uses the *conditional Or*, the rate limit will be shared between the URIs. Create different rules if you wish to have a Rate Limite for each URI.

:::note
The `Maximum Burst size` is only available for the `Req/s` (request per second).
:::

### Set WAF Rule Set 

<Tag severity="info" client:only="vue">Requires WAF</Tag>

It associates the WAF Rule Set to be used in the request. WAF policies must be previously configured in **Edge Libraries** > **WAF Rules**. You must also define the **WAF mode**: **Learning** or **Blocking**. Read more on [WAF Rule Sets](/en/documentation/products/secure/edge-firewall/web-application-firewall/rules-set/). 

:::caution[Warning]
Azion platform maintains only one `Set WAF Rule Set` behavior configuration for each criteria logic. If you have two different edge firewalls configured with the same criteria logic but different `Set WAF Rule Set` behaviors, only the most recent one will be processed. This can be useful in case your application needs constant switching between two or more behaviors for the same criteria.
:::

> ### Run a Function

Author: ```suggestion
### Run Function
```

<Tag severity="info" client:only="vue">Requires Edge Functions</Tag>

It runs a function specified as a parameter. The function must have been previously [instantiated and parameterized](/en/documentation/products/guides/secure/instantiate-edge-functions/) in the Functions tab in order to be used.

### Set Custom Response

It allows a customized response when the request matches the criteria. You can customize the `Status Code` by changing it from 200 to 499, the `Content Type` header, and `Content Body` of your request with a maximum of `500` characters.

### Behaviors execution and logic

<RulesExecution />

---

## Debugging rules

> You can [debug rules created with Rules Engine](/en/documentation/products/guides/debug-rules/) created in Rules Engine for Edge Firewall through the GraphQL API, Data Stream, or Real-Time Events.

Author: ```suggestion
You can [debug rules created with Rules Engine for Edge Firewall](/en/documentation/products/guides/debug-rules/) through the GraphQL API, Data Stream, or Real-Time Events.
```
